fix(sunshine): make cuda and unit tests optional

FineWolf opened the pull request:
<!---
    NOTE: DO NOT DELETE ANY COMMENTS OR SECTIONS OF THIS TEMPLATE.
    THEY ARE IMPORTANT FOR THE MAINTAINERS.
    IT IS OKAY TO LEAVE SECTIONS EMPTY AND BOXES UNCHECKED IF THEY DO NOT APPLY TO YOUR PR.
--->

## Description

The current published PKGBUILD (`2025.628.4510-5`) doesn't build on Arch currently due to the missing `boost` makedepends. There are also a few issues with the package itself that causes issues for non Nvidia systems.

A few changes have been made in this pull request.

### Quality of life changes for you

- I moved `_commit` into a variable so that, when you want to release an update, you don't have 3 places where you have to change the hash. This makes maintaining the PKGBUILD easier, and it also makes adjusting your pipelines dynamically to use a different commit easier as well (just set the `_commit` variable).
- I made sure to pass `-j$(nproc)` to speed up building.
- I updated the `.gitignore` files so that build artifacts are ignored.

### Make `cuda` optional for building

Requiring `cuda` for building on non-Nvidia systems is quite a heavy requirement. Even if it is an `optdepends`, for a build it still will be required. Since `cuda` is huge (4.8 GB installed, ~2 GB download), and won't realistically be used on AMD systems, I added some logic to make it truly optional at build time. If `cuda` or `nvidia-utils` is detected as installed at build time, the build will require `cuda`. If not, `-DCUDA_FAIL_ON_MISSING=OFF` is passed. You can pass `_use_cuda=true` to force it on at any time.

I've adjusted the GitHub pipelines to build with `cuda` regardless, so that your CI processes won't be impacted, and your builds will have `cuda` enabled by default.

### Make `xorg-server-xvfb` optional for building

Same comment here for `xorg-server-xvfb`, the end-users don't really need it; it is however super useful for testing, so I made it an optional component of the build. Your pipelines have been adjusted so that it is present in your builds as an `optdepends`.

### Make tests failure optional when a user builds it

At the moment, the tests are failing after a build. I tried to debug it a little, but I wasn't able to identify the cause, and the resulting package still works fine on the two systems I tested them on.

The test failures all seem to be in the assertion for `platf_deinit`.

```
[  SKIPPED ] EncoderVariants/EncoderTest.ValidateEncoder/software (0 ms)
/home/amoore/Projects/OSS/pacman-repo/pkgbuilds/sunshine/src/sunshine/tests/unit/../tests_common.h:20: Failure
Value of: platf_deinit
  Actual: false
Expected: true

[...]

 0 FAILED TESTS
[  FAILED  ] Configurations/AudioTest: SetUpTestSuite or TearDownTestSuite
[  FAILED  ] MouseInputs/MouseHIDTest: SetUpTestSuite or TearDownTestSuite
[  FAILED  ] EncoderVariants/EncoderTest: SetUpTestSuite or TearDownTestSuite
```

Either way, it's safe to assume that if a release was made by you, that release was tested previously. Just like the community PKGBUILD beforehand was doing, I made failing on tests optional. The GitHub CI pipelines were adjusted to fail if the test fail, however, because that's quite an important metric to have during a CI build.

### ~~Restore `boost` 1.88 fixes from community PKGBUILD~~

~~Arch currently ships with [`boost` 1.88](https://archlinux.org/packages/extra/x86_64/boost/)  which unfortunately requires fixes in the main package. For now, to restore functionality, the fixes that were present in the community provided PKGBUILD were ported to this PKGBUILD.~~

~~There's guidance on how to possibly fix this in the upstream bug report: https://github.com/boostorg/process/issues/480~~

### Correctness fix in sunshine.install

`sunshine.install` was modified to use `which`. That may cause undesired behaviour if there is a different `sunshine` binary present higher in the user's `PATH`. The `sunshine.install` file was modified to use the proper installation path instead, which avoids that problem.

### Screenshot
<!--- Include screenshots if the changes are UI-related. --->


### Issues Fixed or Closed
<!--- If this PR fixes or closes any issues, please list them here. --->
<!--- Close issue example: `- Closes #1` --->
<!--- Fix bug issue example: `- Fixes #2` --->
<!--- Resolve issue example: `- Resolves #3` --->


#### Roadmap Issues
<!--- If this PR solves any roadmap issues (https://github.com/LizardByte/roadmap/issues), please list them here. --->
<!--- e.g. `- Closes https://github.com/LizardByte/roadmap/issues/1` --->


## Type of Change
<!--- Select the type of change your PR introduces. You can select multiple types. --->
- [ ] **feat**: New feature (non-breaking change which adds functionality)
- [x] **fix**: Bug fix (non-breaking change which fixes an issue)
- [ ] **docs**: Documentation only changes
- [ ] **style**: Changes that do not affect the meaning of the code (white-space, formatting, missing semicolons, etc.)
- [ ] **refactor**: Code change that neither fixes a bug nor adds a feature
- [ ] **perf**: Code change that improves performance
- [ ] **test**: Adding missing tests or correcting existing tests
- [ ] **build**: Changes that affect the build system or external dependencies
- [ ] **ci**: Changes to CI configuration files and scripts
- [ ] **chore**: Other changes that don't modify src or test files
- [ ] **revert**: Reverts a previous commit
- [ ] **BREAKING CHANGE**: Introduces a breaking change (can be combined with any type above)


## Checklist
<!--- Please check all applicable boxes. If a box does not apply, leave it unchecked. --->
- [ ] Code follows the style guidelines of this project
- [x] Code has been self-reviewed
- [ ] Code has been commented, particularly in hard-to-understand areas
- [x] Code docstring/documentation-blocks for new or existing methods/components have been added or updated
- [ ] Unit tests have been added or updated for any new or modified functionality

### AI Usage
<!--- Select the option that best describes AI usage in this PR (choose only one). --->
- [x] **None**: No AI tools were used in creating this PR
- [ ] **Light**: AI provided minor assistance (formatting, simple suggestions)
- [ ] **Moderate**: AI helped with code generation or debugging specific parts
- [ ] **Heavy**: AI generated most or all of the code changes

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/LizardByte/pacman-repo?pullRequest=52) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/LizardByte/pacman-repo?pullRequest=52) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/LizardByte/pacman-repo?pullRequest=52) it.</sub>

ReenigneArcher commented on .gitignore:
```diff
@@ -1,2 +1,8 @@
 .idea/
+
+# ignore build files
+/pkgbuilds/sunshine/pkg
+/pkgbuilds/sunshine/src
+/pkgbuilds/sunshine/sunshine
+/pkgbuilds/sunshine/*.pkg.*
```
```suggestion
/pkgbuilds/*/pkg
/pkgbuilds/*/src
/pkgbuilds/*/sunshine
/pkgbuilds/*/*.pkg.*
```

To support future packages without having to explicitly add all 4 entries for each one.

ReenigneArcher commented on pkgbuilds/sunshine/.SRCINFO:
```diff
@@ -38,9 +40,10 @@ pkgbase = sunshine
-	optdepends = cuda: Nvidia GPU encoding support
 	optdepends = libva-mesa-driver: AMD GPU encoding support
-	optdepends = xorg-server-xvfb: Virtual X server for headless testing
+	optdepends = intel-media-driver: Intel GPU encoding support
```
This is not used anymore: https://github.com/LizardByte/Sunshine/pull/2681

ReenigneArcher commented on pkgbuilds/sunshine/.gitignore:
The `.gitignore` from the root is copied to create the `.gitignore` in the AUR repo.

Perhaps put this in the repo root and rename to `.gitignore-aur`, then update https://github.com/LizardByte/pacman-repo/blob/9bdcdf074050079b21a6c18829a1b7d5d8dc3c44/.github/workflows/sync-aur.yml#L126

ReenigneArcher commented on pkgbuilds/sunshine/PKGBUILD:
```diff
@@ -50,62 +58,144 @@ makedepends=(
-  'cuda: Nvidia GPU encoding support'
   'libva-mesa-driver: AMD GPU encoding support'
-  'xorg-server-xvfb: Virtual X server for headless testing'
+  'intel-media-driver: Intel GPU encoding support'
```
same comment

ReenigneArcher commented on pkgbuilds/sunshine/.SRCINFO:
```diff
@@ -38,9 +40,10 @@ pkgbase = sunshine
 	optdepends = libva-mesa-driver: AMD GPU encoding support
-	optdepends = xorg-server-xvfb: Virtual X server for headless testing
+	optdepends = intel-media-driver: Intel GPU encoding support
+	optdepends = cuda: Nvidia GPU encoding support
```
These should probably be alphabetical... looks like things were re-ordered in several places, which should probably be undone. e.g. `makedepends = cuda`

ReenigneArcher commented on pkgbuilds/sunshine/PKGBUILD:
```diff
@@ -50,62 +58,144 @@ makedepends=(
 
     cd "${srcdir}/build/tests"
-    ./test_sunshine --gtest_color=yes
+    if [[ "${_allow_test_failure::1}" == "t" ]]; then
```
Would it be better to have an option to not run the gtest suite at all? This is mostly for CI and can definitely understand AUR users not wanting to run them. An extremely simple test could be added like just running `sunshine --version`?

ReenigneArcher reviewed (changes requested):
Thank you for the PR! I understand the points you're mentioning and agree they should be improved, but I have some concerns about merging this as is.

1. Boost 1.87 is fetched during the cmake configure process. Sunshine uses boost 1.87 for all builds currently. I'd like to move to the newer version, but haven't investigated migrating, so thank you for the link. I think I'm okay with the boost hack for the stable here, but once we have upstream updated to 1.88. Would you be interested in helping us migrate upstream to 1.88?
2. If this were merged as is, changes won't persist on the next stable release of Sunshine. PR(s) will need to be made upstream as well. https://github.com/LizardByte/Sunshine/tree/master/packaging/linux/Arch

FineWolf commented:
> Would you be interested in helping us migrate upstream to 1.88?

For boost-1.88, I believe the change is just making sure the `v1` headers are used for now before migrating to v2 for the process library. Unfortunately, 1.88 removed `<boost/process/v1.hpp>`. It was later restored in 1.89.

I didn't like the way this was made in the community PKGFILE, so instead I included a patch file that is applied on top of the upstream repo.

The patch can be applied upstream as well.

> The .gitignore from the root is copied to create the .gitignore in the AUR repo.

It is, but the root of the repository isn't at the same level, so copying the `.gitignore` directly won't work. I adjusted the pipeline to use `sed` and adjust the path roots automatically.

> 'intel-media-driver: Intel GPU encoding support'

Removed :)

> Would it be better to have an option to not run the gtest suite at all?

Absolutely. I changed the setting to make _building and running_ the tests optional. They are off by default. The pipelines were adjusted to turn them on.

ReenigneArcher commented:
> For boost-1.88, I believe the change is just making sure the v1 headers are used for now before migrating to v2 for the process library. Unfortunately, 1.88 removed <boost/process/v1.hpp>. It was later restored in 1.89.

I opened a PR to update to 1.89 upstream: https://github.com/LizardByte/Sunshine/pull/4253 ... It's still missing a little bit. For some reason it fails on Arch, though it succeeds in Debian, Fedora, and Ubuntu.

> I didn't like the way this was made in the community PKGFILE, so instead I included a patch file that is applied on top of the upstream repo.

I agree, the patch file is nicer!

> Do you want me to mirror my changes and submit a pull request in that repo as well?

Yes, that would be ideal so everything isn't lost when we make the next stable release.

> It is, but the root of the repository isn't at the same level, so copying the .gitignore directly won't work. I adjusted the pipeline to use sed and adjust the path roots automatically.

Oh, I meant you could have had a separate `.gitignore-aur` file at the root, and then just copy that, instead of the original `.gitignore`.

FineWolf commented:
> Yes, that would be ideal so everything isn't lost when we make the next stable release.

Done, see https://github.com/LizardByte/Sunshine/pull/4254

> Oh, I meant you could have had a separate `.gitignore-aur` file at the root, and then just copy that, instead of the original `.gitignore`.

That's way simpler. I adjusted my pull request.

ReenigneArcher commented on pkgbuilds/sunshine/PKGBUILD:
```diff
@@ -39,9 +47,9 @@ depends=(
 makedepends=(
   'appstream'
   'appstream-glib'
+  'boost'
```
Got it, I think boost can be removed from dependencies then. The build won't use the provided by Arch.

FineWolf commented:
Removed trailing line following `lint` failure.
